backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepositoryFactory.java

```diff
@@ -146,21 +146,33 @@ IGitMacheteRepository createGitMacheteRepository(IBranchLayout branchLayout) thr
 
       Map<String, List<IGitCoreReflogEntry>> filteredReflogByRemoteTrackingBranchName = remoteTrackingBranches
           .toMap(
-              /* keyMapper */ remoteBranch -> remoteBranch.getShortName(),
+              /* keyMapper */ remoteBranch -> remoteBranch.getName(),
```

Comment:
This line is the core of the fix

Reply:
"short name" used to be e.g. just `master` for `origin/master`, which clashed with `master` the local branch :frowning_face: